fdb-record-layer-core/src/test/proto/test_records_2.proto

```diff
@@ -33,6 +34,11 @@ message MyLongRecord {
   optional bytes bytes_value = 2;
 }
 
+message MyUuidRecord {
```

Comment:
1. Can you make a test based on the primary key being a UUID?
2. Can you create indexes using an UUID?

Reply:
Added a couple of tests in record-layer-core. There are also end-to-end tests in yaml framework which explicitly tests UUID as a primary key and in Indexes.